python/cudf/cudf/utils/utils.py

```diff
@@ -89,7 +89,12 @@ def scalar_broadcast_to(scalar, size, dtype=None):
         return column.column_empty(size, dtype=dtype, masked=True)
 
     if isinstance(scalar, pd.Categorical):
-        return scalar_broadcast_to(scalar.categories[0], size).astype(dtype)
+        if dtype is None:
+            return create_cat_column_from_pd_cat_scalar(scalar, size)
```

Comment:
Please see my note below before accepting/not accepting this suggestion.

```suggestion
            return _categorical_scalar_broadcast_to(scalar, size)
```

Reply:
Resolved with commit https://github.com/rapidsai/cudf/commit/b479d60762c81d1ffbe6723bd7d1910834318f5a